src/api.js

```diff
@@ -1,29 +1,42 @@
 import Bottleneck from 'bottleneck'
 
-import { betterAtob, sortContents } from '@/utils'
-import RelativeTime from '@/github-relative-time'
 import { getState } from '@/storage'
+import { betterAtob, sortContents, removeToken } from '@/utils'
+import RelativeTime from '@/github-relative-time'
 
 const MAX_REQUESTS = 10
 
 const limiter = new Bottleneck({
   maxConcurrent: MAX_REQUESTS,
 })
 
-const githubFetch = (fragment, { importance, ...options } = {}) =>
-  fetch(`https://api.github.com/${fragment}`, {
+const githubFetch = (fragment, { importance, ...options } = {}) => {
+  const { token } = getState()
+
+  return fetch(`https://api.github.com/${fragment}`, {
     ...options,
     importance,
     headers: {
-      Authorization: `token ${getState().token}`,
+      Authorization: `token ${token}`,
     },
   }).then(response => {
     if (response.status < 200 || response.status > 299) {
+      // A user may have manually removed token from GitHub account,
+      // if they have, this should prompt them for a new token
+      if (response.status === 401 && token) {
+        removeToken().then(() => {
+          throw new Error(
+            `${response.status}: ([tako] It is possible the associated token has been revoked) ${response.statusText}`
+          )
+        })
+      }
+
```

Comment:
Neat! This should go into a separate PR though, makes it easier to review when it's not packed into a whole bundle of changes. Also, You're removing the token and then throwing an error, did you test this state? I'm not sure whether we can see ErrorBoundary and AskForToken at the same time.

Reply:
So you want this bit of code removed (for this PR)?  Keep in mind, this code resolves the issue of not being prompted for a new token.

Yes, I did test this state - it throws the appropriate error.

Reply:
>So you want this bit of code removed (for this PR)? Keep in mind, this code resolves the issue of not being prompted for a new token.

Sure! It's a separate topic and a separate fix

Reply:
Sounds good. I'll get it removed. If/when this PR gets merged I can submit a separate PR for that if you want me to.  I can make it so the user gets an error stating the token may have been removed. Only after the user has clicked 'Try Again' it will promote for a new token. Just an idea...

I can write it however you prefer.

Reply:
Is it possible for you to pull out the changes you needed to do and open an additional PR?
This way, we could get the expired-token-handling in before dealing with all this ;)

Reply:
I am confused as to what you want exactly. 

You want me to separate the Firefox changes from the expired token handling changes and submit 2 separate PRs, back to back?....

Reply:
Yes my dude. This PR is about _Firefox Integration_. Yet here we are spending time on _a different feature, which prevents errors when people revoke their token_. These are two distinct topics and concerns. Had you not put both of those into one PR, we could have gotten the latter one merged already, cause it's a nice and easy change. Instead, it's bunched together with wide-reaching changes that I don't want to merge like this.

Reply:
There can be multiple open PRs per topic. It's very useful!

Reply:
All remnants of the 'expired token handling' have been removed from this PR.

Reply:
cc @#20